### PR TITLE
Update to asl-help.

### PR DESCRIPTION
**Pull request by ero-qt**
This pull request is part of an effort to update existing auto splitters to use `asl-help`.

I implore you to please, please, please test the script thoroughly as I've only made sure it passes the `TryLoad` along with minor checks (like starting and load removal).

Check the corresponding PR to update the library in the `LiveSplit.AutoSplitters.xml` at https://github.com/LiveSplit/LiveSplit.AutoSplitters/pull/999.

### CODE REVIEW

File: Escape Academy/escapeacademy.asl [hunk 1]
```diff
@@ -2,63 +2,40 @@ state("Escape Academy") { }
 
 startup
 {
-	var bytes = File.ReadAllBytes(@"Components\LiveSplit.ASLHelper.bin");
-	var type = Assembly.Load(bytes).GetType("ASLHelper.Unity");
-	vars.Helper = Activator.CreateInstance(type, timer, this);
+	Assembly.Load(File.ReadAllBytes("Components/asl-help")).CreateInstance("Unity");
 
 	vars.Helper.AlertLoadless();
```
Comment (mitchell-merry) [inline Escape Academy/escapeacademy.asl:7]: Getting this error here:
```
LiveSplit.exe Error: 0 : 
Exception thrown: 'System.Reflection.TargetInvocationException' in 'startup' method:
Exception has been thrown by the target of an invocation.

   at ASL line 7 in 'startup'

   at LiveSplit.ASL.ASLMethod.Call(LiveSplitState timer, ExpandoObject vars, String& version, Double& refreshRate, Object settings, ExpandoObject old, ExpandoObject current, Process game) 
   at LiveSplit.ASL.ASLScript.RunStartup(LiveSplitState state) 
   at LiveSplit.UI.Components.ASLComponent.LoadScript() 
```
Occurs on line 9 if this is commented out.

File: Escape Academy/escapeacademy.asl [hunk 2]
```diff
@@ -2,63 +2,40 @@ state("Escape Academy") { }
...
 
-		vars.Helper["currentMGState"] = mgm.Make<int>("_instance", "_currentMetaGameState");
-		vars.Helper["_asyncLoadingLevel"] = mgm.Make<bool>("_instance", "sceneLevelManager", slm["_asyncLoadingLevel"]);
-
-		var lsm = mono.GetClass("Escape.Metagame.LevelSelect.LevelSelectManager");
-		vars.Helper["LSState"] = lsm.Make<int>("Instance", "_currentLevelSelectState");
+		var lsm = mono["Escape.Metagame.LevelSelect.LevelSelectManager"];
+		vars.Helper["LSState"] = mono.Make<int>(lsm, "Instance", "_currentLevelSelectState");
 
 		var rm = mono.GetClass("Escape.Rooms.RoomManager");
```
Comment (mitchell-merry) [inline Escape Academy/escapeacademy.asl]: Should this be:
```suggestion
		var rm = mono["Escape.Rooms.RoomManager"];
```

File: Escape Academy/escapeacademy.asl (hunk 1 above)
Comment (mitchell-merry) [reply Escape Academy/escapeacademy.asl:7]: Not sure if this is a me problem or not, but I don't have the time this morning to diagnose it :P

File: Haunted Gas Station/hgs.asl [hunk 3]
```diff
@@ -2,37 +2,26 @@ state("Haunted Gas Station") { }
 
 startup
 {
-	var bytes = File.ReadAllBytes(@"Components\LiveSplit.ASLHelper.bin");
-	var type = Assembly.Load(bytes).GetType("ASLHelper.Unity");
-	vars.Helper = Activator.CreateInstance(type, timer, this);
+	Assembly.Load(File.ReadAllBytes("Components/asl-help")).CreateInstance("Unity");
+	vars.Helper.GameName = "HGS";
```
Comment (mitchell-merry) [inline Haunted Gas Station/hgs.asl:6]: Also getting the `'System.Reflection.TargetInvocationException'` here.

**Review by mitchell-merry — CHANGES_REQUESTED** [review]
loving the changes to the code structure! just can't get either script to run past startup, and I don't really have the time this morning to diagnose further.

I'll be happy to update the rest of my autosplitters to the new library if you wanted to focus your development efforts on switching to the new one elsewhere btw. Thanks!

File: Escape Academy/escapeacademy.asl (hunk 2 above)
Comment (ero-qt) [reply Escape Academy/escapeacademy.asl]: Fixed in 18480731f5d3a0d49b13a99a8574552ae87de055.

**Pull request merged**